### ISSUE TITLE
Mapping nested object structures confuses AgileMapper

### ISSUE THREAD

**Issue by bslbckr** [agileobjects/AgileMapper#125]
We are facing again some problems with AgileMapper. When we are mapping nested object structures - in this specific case the failure arises at nesting level four - Something confuses AgileMapper in it starts trying to map things on each other that dont even exist. Consequently we are receiving RuntimeExceptions in this scenario.

The StackTrace looks as follows:
```
[AgileObjects.AgileMapper.MappingException: An exception occurred mapping ParamSet -> ParamSet with rule set CreateNew.]
  [AgileObjects.AgileMapper.MappingException: An exception occurred mapping ParamSet.ParamObj -> ParamSet.po with rule set CreateNew.]
    [AgileObjects.AgileMapper.MappingException: An exception occurred mapping ParamSet.ParamObj.Collection -> ParamSet.po.Collection with rule set CreateNew.]
      [AgileObjects.AgileMapper.MappingException: An exception occurred mapping ParamSet.Collection.Children -> ParamSet.po.Collection.Children with rule set CreateNew.]
```

The last step is not working, as `ParmSet.Collection` does not exists. 
The mapping configuration we use, is as follows:
```
mapper.WhenMapping.From<Source.ParamSet>().To<Target.ParamSet>().Map(ctxt => ctxt.Source.ParamObj).ToCtor<Target.ParamObj>().And.Ignore(t => t.ParamObj);
```

I could imagine that it relates to the ignore-clause, however this is just because ParamObj is missing in sequence.

In addition to this, more or less contextless, issue I create a fiddle to illustrate my problem:
https://dotnetfiddle.net/oBdpdS

**Comment by SteveWilkes** [comment]
I think 'ParamSet.Collection.Children' is a red herring - this is to do with it mapping nested members of type `object`.

I'll take a look and sort it out. Thanks! :)

**Comment by SteveWilkes** [comment]
Just to note - the Mapper is falling over trying to generate a mapping plan for `Source.ParamSet.Collection` -> `Target.ParamSet.Collection.ParamObj`, which it's paired up because the source Types don't have a `ParamObj` property on `ParamCol`. The actual mapping works fine, just generating the plan object doesn't.

Incidentally, if Target.ParamObj's constructor argument is renamed to 'paramObj', no configuration is necessary. AgileMapper knows to map `Source.ParamSet.ParamObj` to it, and won't map to the `Target.ParamSet.ParamObj` property as it has a matching constructor argument :)

**Comment by bslbckr** [comment]
Thanks for the clarification. I was already doubting my own senses. I had already seen the mapping working, and yesterday I saw it failing reproducably. You can guess now, what my last change was? :-) 

Thanks once again for your effort! 👍

**Comment by SteveWilkes** [comment]
You're welcome! :)

This is fixed in the latest code, release to follow.

Cheers!

**Comment by bslbckr** [comment]
> Just to note, this bug only occurs when a mapping plan is generated up-front - if we let the mapper lazy-load the mapping plan it works fine.

But only the first time. I skipped creation of execution plans in order to work around this issue.  But it then bite me the second time I had to use the mapper.

**Comment by SteveWilkes** [comment]
This is fixed in v1.3, now available [on NuGet](https://www.nuget.org/packages/AgileObjects.AgileMapper/). Cheers! :)
